drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/PluginHandler.java

```diff
@@ -0,0 +1,135 @@
+                         final ManagedInstance<BRLConditionColumnPlugin> brlConditionColumnPlugin,
+                         final ManagedInstance<ConditionColumnPlugin> conditionColumnPlugin,
+                         final ManagedInstance<ActionRetractFactPlugin> actionRetractFactPlugin,
+                         final ManagedInstance<ActionSetFactPlugin> actionSetFactPlugin,
+                         final ManagedInstance<ActionWorkItemSetFieldPlugin> actionWorkItemSetFieldPlugin,
+                         final ManagedInstance<ActionWorkItemPlugin> actionWorkItemPlugin,
+                         final ManagedInstance<BRLActionColumnPlugin> brlActionColumnPlugin) {
+
+        this.wizardManagedInstance = wizardManagedInstance;
+        this.brlConditionColumnPlugin = brlConditionColumnPlugin;
+        this.conditionColumnPlugin = conditionColumnPlugin;
+        this.actionRetractFactPlugin = actionRetractFactPlugin;
+        this.actionSetFactPlugin = actionSetFactPlugin;
+        this.actionWorkItemSetFieldPlugin = actionWorkItemSetFieldPlugin;
+        this.actionWorkItemPlugin = actionWorkItemPlugin;
+        this.brlActionColumnPlugin = brlActionColumnPlugin;
+    }
+
+    public void init(final GuidedDecisionTableView.Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    public void edit(final BRLConditionColumn column) {
+        final BRLConditionColumnPlugin plugin = brlConditionColumnPlugin.get();
+
+        plugin.setOriginalColumnConfig52(column);
+
+        openWizard(plugin);
+    }
+
+    public void edit(final ActionCol52 column) {
+        final DecisionTableColumnPlugin plugin;
+
+        if (column instanceof ActionWorkItemSetFieldCol52 || column instanceof ActionWorkItemInsertFactCol52) {
+            plugin = actionWorkItemSetFieldPlugin.get();
+        } else if (column instanceof ActionInsertFactCol52 || column instanceof ActionSetFieldCol52) {
+            plugin = actionSetFactPlugin.get();
+        } else if (column instanceof ActionRetractFactCol52) {
+            plugin = actionRetractFactPlugin.get();
+        } else if (column instanceof ActionWorkItemCol52) {
+            plugin = actionWorkItemPlugin.get();
+        } else if (column instanceof LimitedEntryBRLActionColumn) {
+            plugin = brlActionColumnPlugin.get();
+        } else if (column instanceof BRLActionColumn) {
+            plugin = brlActionColumnPlugin.get();
+        } else {
+            return;
+        }
+
+        plugin.setOriginalColumnConfig52(column);
+
+        openWizard(plugin);
+    }
+
+    public void edit(final Pattern52 pattern,
+                     final ConditionCol52 column) {
+        final ConditionColumnPlugin plugin = conditionColumnPlugin.get();
+
+        plugin.setOriginalPattern52(pattern);
+        plugin.setOriginalColumnConfig52(column);
+
+        openWizard(plugin);
+    }
+
+    void openWizard(final DecisionTableColumnPlugin plugin) {
+        if (presenter.isReadOnly()) {
```

Comment:
Precisely. This is just a safe-guard.

All "newColumnX" had this check, so I think that would be reasonable to keep this verification. However, we can remove this, if you consider that it's redundant.